stan/math/prim/prob/discrete_range_cdf.hpp

```diff
@@ -55,17 +55,16 @@ double discrete_range_cdf(const T_y& y, const T_lower& lower,
     if (y_dbl < lower_vec[n]) {
       return 0;
     }
-    if (y_dbl > upper_vec[n]) {
-      return 1;
-    }
   }
 
   double cdf(1.0);
   for (size_t n = 0; n < N; n++) {
     const double y_dbl = y_vec[n];
-    const double lower_dbl = lower_vec[n];
-    const double upper_dbl = upper_vec[n];
-    cdf *= (y_dbl - lower_dbl + 1) / (upper_dbl - lower_dbl + 1);
+    if (y_dbl < upper_vec[n]) {
```

Comment:
Apologies but for the discrete range changes you're just gonna have to check me.

This if goes inside this loop because y_dbl > upper_vec multiplies the cdf by 1.0 (the vectorized cdfs are the products of all the constituent cdfs).

If the if statement stays outside, then this function erroneously returns 1 if any single cdf term should be one.